types/picomatch/lib/picomatch.d.ts

```diff
@@ -0,0 +1,250 @@
+import constantsImport = require('./constants');
+import parse = require('./parse');
+import parseImport = require('./parse');
+import scanImport = require('./scan');
+
+/**
+ * Creates a matcher function from one or more glob patterns. The
+ * returned function takes a string to match as its first argument,
+ * and returns true if the string is a match. The returned matcher
+ * function also takes a boolean as the second argument that, when true,
+ * returns an object with additional information.
+ *
+ * ```js
+ * const picomatch = require('picomatch');
+ * // picomatch(glob[, options]);
+ *
+ * const isMatch = picomatch('*.!(*a)');
+ * console.log(isMatch('a.a')); //=> false
+ * console.log(isMatch('a.b')); //=> true
+ * ```
+ * @param glob One or more glob patterns.
+ * @return Returns a matcher function.
+ * @api public
+ */
+declare function picomatch<T extends true | false>(
+    glob: string | string[],
+    options?: picomatch.PicomatchOptions,
+    returnState?: T,
+): T extends true ? picomatch.MatcherWithState : picomatch.Matcher;
+
+declare namespace picomatch {
+    interface Matcher {
+        (test: string): boolean;
+    }
+
+    interface MatcherWithState extends Matcher {
+        state: parse.ParseState;
```

Comment:
This lets you remove the duplicate import for `./parse`, above.

```suggestion
        state: parseImport.ParseState;
```